lapce-proxy/src/plugin/wasi.rs

```diff
@@ -183,37 +186,37 @@ impl PluginServerHandler for Plugin {
 
 impl Plugin {
     fn initialize(&mut self) {
+        let server_rpc = self.host.server_rpc.clone();
         let workspace = self.host.workspace.clone();
         let configurations = self.configurations.as_ref().map(unflatten_map);
-        let root_uri = workspace.map(|p| Url::from_directory_path(p).unwrap());
-        if let Ok(value) = self.host.server_rpc.server_request(
-            Initialize::METHOD,
-            #[allow(deprecated)]
-            InitializeParams {
-                process_id: Some(process::id()),
-                root_path: None,
-                root_uri,
-                capabilities: client_capabilities(),
-                trace: None,
-                client_info: None,
-                locale: None,
-                initialization_options: configurations,
-                workspace_folders: None,
-            },
-            None,
-            None,
-            false,
-        ) {
-            let result: InitializeResult = serde_json::from_value(value).unwrap();
-            self.host.server_capabilities = result.capabilities;
-            self.host.server_rpc.server_notification(
-                Initialized::METHOD,
-                InitializedParams {},
+        let server_capabilities = self.host.server_initialisation.clone();
+        thread::spawn(move || {
```

Comment:
All the changes here are necessary to make offloading the init request here to a separate thread.